gateway/mw_graphql.go

```diff
@@ -67,145 +56,65 @@ func (m *GraphQLMiddleware) Init() {
 		log.Errorf("Schema normalization was not successful. Reason: %v", normalizationResult.Errors)
 	}
 
-	m.Spec.GraphQLExecutor.Schema = schema
-
-	if needsGraphQLExecutionEngine(m.Spec) {
-		absLogger := abstractlogger.NewLogrusLogger(log, absLoggerLevel(log.Level))
-		m.Spec.GraphQLExecutor.Client = &http.Client{
-			Transport: &http.Transport{TLSClientConfig: tlsClientConfig(m.Spec, nil)},
-		}
-		m.Spec.GraphQLExecutor.StreamingClient = &http.Client{
-			Timeout:   0,
-			Transport: &http.Transport{TLSClientConfig: tlsClientConfig(m.Spec, nil)},
-		}
-
-		if m.Spec.GraphQL.Version == apidef.GraphQLConfigVersionNone || m.Spec.GraphQL.Version == apidef.GraphQLConfigVersion1 {
-			m.initGraphQLEngineV1(absLogger)
-		} else if m.Spec.GraphQL.Version == apidef.GraphQLConfigVersion2 {
-			m.initGraphQLEngineV2(absLogger)
-		} else {
-			log.Errorf("Could not init GraphQL middleware: invalid config version provided: %s", m.Spec.GraphQL.Version)
-		}
+	reusableBodyReadCloser := func(buf io.ReadCloser) (io.ReadCloser, error) {
+		return newNopCloserBuffer(buf)
 	}
-}
 
-func (m *GraphQLMiddleware) initGraphQLEngineV1(logger *abstractlogger.LogrusLogger) {
-	typeFieldConfigurations := m.Spec.GraphQL.TypeFieldConfigurations
-	if m.Spec.GraphQLExecutor.Schema.HasQueryType() {
-		typeFieldConfigurations = append(typeFieldConfigurations, datasource.TypeFieldConfiguration{
-			TypeName:  m.Spec.GraphQLExecutor.Schema.QueryTypeName(),
-			FieldName: "__schema",
-			DataSource: datasource.SourceConfig{
-				Name: SchemaDataSource,
-				Config: func() json.RawMessage {
-					res, _ := json.Marshal(datasource.SchemaDataSourcePlannerConfig{})
-					return res
-				}(),
+	if m.Spec.GraphQL.Version == apidef.GraphQLConfigVersionNone || m.Spec.GraphQL.Version == apidef.GraphQLConfigVersion1 {
+		m.Spec.GraphEngine, err = graphengine.NewEngineV1(graphengine.EngineV1Options{
```

Comment:
Can we add an info-level log regarding the deprecation of engine version 1? It can also be good to warn the user, if `graphql.version` is an empty string. In this case, we are silently setting `graphql.version` to 1.

Reply:
added!